flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/JobManagerDeploymentStatus.java

```diff
@@ -37,7 +37,11 @@
     DEPLOYING,
 
     /** JobManager deployment not found, probably not started or killed by user. */
-    MISSING;
+    // TODO: currently a mix of SUSPENDED and ERROR, needs cleanup
```

Comment:
Does this need to add the `SUSPENDED` status?

Reply:
I think yes, but outside of this PR. We should get rid of MISSING also.